consensus/src/experimental/buffer_manager.rs

```diff
@@ -413,24 +450,30 @@ impl BufferManager {
             // it is possible that we already signed this buffer item (double check after the final integration)
             if item.is_executed() {
                 // we have found the buffer item
-                let signed_item = item.advance_to_signed(self.author, signature);
-                let maybe_proposer = signed_item
-                    .unwrap_signed_ref()
+                let mut signed_item = item.advance_to_signed(self.author, signature);
+                let signed_item_mut = signed_item.unwrap_signed_mut();
+                let maybe_proposer = signed_item_mut
                     .executed_blocks
                     .last()
                     .unwrap()
                     .block()
                     .author();
-                let commit_vote = signed_item.unwrap_signed_ref().commit_vote.clone();
+                let commit_vote = signed_item_mut.commit_vote.clone();
 
-                self.buffer.set(&current_cursor, signed_item);
                 if let Some(proposer) = maybe_proposer {
-                    self.commit_msg_tx
-                        .send_commit_vote(commit_vote, proposer)
-                        .await;
+                    let sender = self.commit_msg_tx.clone();
+                    tokio::spawn(async move {
+                        if let Err(e) = sender.send_commit_vote(commit_vote, proposer).await {
+                            warn!("Failed to send commit vote {:?}", e);
+                        }
+                    });
                 } else {
-                    self.commit_msg_tx.broadcast_commit_vote(commit_vote).await;
+                    let commit_vote = CommitMessage::Vote(commit_vote);
+                    signed_item_mut
+                        .rb_handle
+                        .replace((Instant::now(), self.do_reliable_broacast(commit_vote)));
```

Comment:
The timestamp seems unused?

Reply:
it's used to decide to re-initiate the rb